src/distribution/multivariate_normal.rs

```diff
@@ -5,76 +5,39 @@ use nalgebra::{Cholesky, Const, DMatrix, DVector, Dim, DimMin, Dyn, OMatrix, OVe
 use std::f64;
 use std::f64::consts::{E, PI};
 
-/// computes both the normalization and exponential argument in the normal distribution
-/// # Errors
-/// will error on dimension mismatch
-pub(super) fn density_normalization_and_exponential<D>(
-    mu: &OVector<f64, D>,
-    cov: &OMatrix<f64, D, D>,
-    precision: &OMatrix<f64, D, D>,
-    x: &OVector<f64, D>,
-) -> std::result::Result<(f64, f64), StatsError>
-where
-    D: DimMin<D, Output = D>,
-    nalgebra::DefaultAllocator: nalgebra::allocator::Allocator<f64, D>
-        + nalgebra::allocator::Allocator<f64, D, D>
-        + nalgebra::allocator::Allocator<(usize, usize), D>,
-{
-    Ok((
-        density_distribution_pdf_const(mu, cov)?,
-        density_distribution_exponential(mu, precision, x)?,
-    ))
-}
-
 /// computes the argument of the exponential term in the normal distribution
-/// ```text
-/// ```
-/// # Errors
-/// will error on dimension mismatch
+/// # Panics
+/// will panic on dimension mismatch
 #[inline]
-pub(super) fn density_distribution_exponential<D>(
+pub(super) fn pdf_exponent_unchecked<D>(
```

Comment:
I'm only skimming the changes at the moment, but the [convention](https://doc.rust-lang.org/stable/std/?search=_unchecked) seems to be that `_unchecked` denotes a function which is `unsafe` and can cause undefined behavior if used incorrectly.

There are *`checked`* functions for number primitives that differ from their non-checked counterparts (`add` vs. `checked_add` which checks for overflow), but functions that panic often don't have a function pre/suffix as far as I know. Maybe there's a place for `pdf_exponent_checked` which returns an `Option`, and then `pdf_exponent` which just uses `pdf_exponent_checked.unwrap()`.

Reply:
Following style from std seems good. I suppose documenting panics is all that's needed. Could also implement `try_... -> Result` instead. Since it's internal and fairly transparent, I'll just have the panicking version.

Reply:
Oops. I missed that these were not exposed to the public API, skimmed too fast.

Doesn't matter that much for non-public API IMO

Reply:
Agreed. Will mention in #284 as well